### PR TITLE
Add select input component

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            labelSelector(b),
+            searchString,
+        ));
+}
+
+export function getLocalISOString() {
+    const tzoffset = (new Date()).getTimezoneOffset() * 60000;
+    const localISOTime = (new Date(Date.now() - tzoffset)).toISOString().slice(0, -1);
+    return localISOTime;
+}
```
Comment (tnagorra) [inline src/utils/index.ts]: Do we need thsi?

File: src/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export const typedMemo: (<T>(c: T) => T) = memo;
```
Comment (tnagorra) [inline src/utils/index.ts]: This we need. Nice!

File: src/index.tsx [hunk 3]
```diff
@@ -16,6 +16,10 @@ export { default as Container } from './components/Container';
 export type { Props as ContainerProps } from './components/Container';
 export { default as ContainerCard } from './components/ContainerCard';
 export type { Props as ContainerCardProps } from './components/ContainerCard';
+export { default as DateInput } from './components/DateInput';
+export type { Props as DateInputProps } from './components/DateInput';
+export { default as DateRangeInput } from './components/DateRangeInput';
+export type { Props as DateRangeInputProps } from './components/DateRangeInput';
 export { default as ElementFragments } from './components/ElementFragments';
```
Comment (tnagorra) [inline src/index.tsx:25]: Do we not need to add SelectInput here?

File: src/utils/index.ts (hunk 1 above)
Comment (frozenhelium) [reply src/utils/index.ts]: Some day

File: src/index.tsx (hunk 3 above)
Comment (frozenhelium) [reply src/index.tsx:25]: Yes we do

File: src/utils/index.ts (hunk 1 above)
Comment (frozenhelium) [reply src/utils/index.ts]: Removed

File: src/components/SearchSelectInput/EmptyOptions/styles.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.empty {
+    padding: var(--tui-spacing-large);
+    color: var(--tui-color-text-disabled-light);
```
Comment (AdityaKhatri) [inline src/components/SearchSelectInput/EmptyOptions/styles.css]: This should be dui

File: src/components/RawInput/index.tsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 import { _cs } from '@togglecorp/fujs';
 
-import type { UiMode } from '../UiModeContext';
-import useUiModeClassName from '../../hooks/useUiModeClassName';
+import type { UiMode } from '#components/UiModeContext';
+import useUiModeClassName from '#hooks/useUiModeClassName';
+import { genericMemo } from '#utils';
```
Comment (AdityaKhatri) [inline src/components/RawInput/index.tsx]: use relative import

File: src/components/SearchSelectInput/EmptyOptions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+function EmptyOptions(props: DefaultEmptyComponentProps) {
+    const {
+        isFiltered = false,
+        optionsPending = false,
+    } = props;
+
+    if (optionsPending) {
+        // FIXME: use loading
```
Comment (AdityaKhatri) [inline src/components/SearchSelectInput/EmptyOptions/index.tsx:17]: @frozenhelium , lets create a new issue in zube for using loading animation here

File: src/components/SearchSelectInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+    listToMap,
+} from '@togglecorp/fujs';
+import { MdCheck } from 'react-icons/md';
+
+import SelectInputContainer, {
+    Props as SelectInputContainerProps,
+} from '#components/SelectInputContainer';
+import ElementFragments from '#components/ElementFragments';
+
+import { genericMemo } from '#utils';
```
Comment (AdityaKhatri) [inline src/components/SearchSelectInput/index.tsx]: use relative import

File: src/components/SelectInputContainer/index.tsx [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+import { IoIosArrowDown, IoIosArrowUp, IoMdClose } from 'react-icons/io';
+
+import GenericSelectInputOption, { ContentBaseProps, OptionKey } from '#components/GenericSelectInputOption';
+import Popup from '../Popup';
+import InputContainer, { Props as InputContainerProps } from '#components/InputContainer';
+import RawInput from '#components/RawInput';
+import Button from '#components/Button';
+import List from '#components/List';
+import useBlurEffect from '#hooks/useBlurEffect';
+import useKeyboard from '#hooks/useKeyboard';
```
Comment (AdityaKhatri) [inline src/components/SelectInputContainer/index.tsx]: remove absolute import

**Review by subinasr — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**